### PR TITLE
feat: gate deploys on Turnkey approval policy coverage

### PR DESCRIPTION
**Pull request by findolor**
## What

Closes RAI-1357.

Depends on #1035.

Adds a read-only Turnkey policy preflight that fails deployment before the running bot is stopped unless every startup MAX approval is immediately authorized.

## Why

A config-valid deployment could replace the running bot and then fail during startup when Turnkey policy did not authorize a required token or wrapper approval. The deployment needs to prove policy coverage against the exact startup approval set while the previous process and installed files are still untouched.

## How

- Share one deterministic approval-target builder between server startup and deployment verification, covering enabled trading or rebalancing assets plus USDC.
- Parse and validate the staged candidate config and secrets without logging credentials.
- Query Turnkey `whoami` and policy-list endpoints through an authenticated read-only client.
- Require the authenticated API user to satisfy allow consensus alone; unknown or multi-approver consensus fails closed.
- Apply Turnkey deny precedence with conservative tri-state applicability: applicable or unprovable denies block coverage.
- Trust explicit target addresses, signing activity fields, and the raw ERC-20 `approve` selector; ABI-derived function fields remain unprovable without interface evidence.
- Package `verify-approvals` with a build-time executable assertion and run it before the single stop boundary.
- Structurally assert that candidate config and secrets are installed only after preflight succeeds, with staged secrets removed by an exit trap.

## Testing

- [x] `nix run .#ci` — 3,579 tests passed, with one configured e2e timing retry; Clippy, rustfmt, DTO regeneration, ESLint, and Svelte checks passed
- [x] `nix develop .#ci-backend -c cargo nextest run -p st0x-evm --all-features policy_client_lists_policy_effects_and_conditions --no-fail-fast`
- [x] `nix develop .#ci-backend -c cargo nextest run -p st0x-hedge --features wallet-turnkey approval_policy --no-fail-fast`
- [x] `nix build .#st0x-liquidity`
- [x] `nix build .#st0x-dashboard`
- [x] `nix eval --json .#deploy.nodes --apply builtins.attrNames`
- [x] `./scripts/validate-deploy-migration.sh`
- [x] `nix develop .#ci-hooks -c pre-commit run --all-files`

## Screenshots

N/A

## Anything else

The verifier never signs or submits an activity and tests use mocked Turnkey query endpoints. The conservative parser is intentional because Turnkey does not expose a policy simulation endpoint.

## Summary by CodeRabbit

* **New Features**
  * Added deployment-time verification that Turnkey approval policies cover all required startup approvals, using staged config/secrets and fail-closed coverage checks.
  * Introduced a candidate-config activation flow that validates before stopping and only then installs the approved artifacts.
  * Added a `verify-approvals` command-line tool to check approval-policy coverage.
* **Bug Fixes**
  * Startup ERC-20 allowance/approval targets now consistently derive from the configured assets (enabled trading/rebalancing equities plus USDC).
* **Documentation**
  * Updated sequencing guidance to reflect the new validation and rollback behavior.

### CODE REVIEW

**Comment by findolor** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/1036?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1036** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/1036?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/1036?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#1035** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/1035?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>add-to-gt-merge-queue</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/ST0x-Technology?ref=merge-queue-instructions-in-stack-comment&prId=1036).

<h2></h2>
<sub>

An organization admin has required the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by findolor** [comment]
@codex Please perform a full code review of this PR.

Review goals:
- Follow all applicable AGENTS.md instructions, including any nested AGENTS.md files that apply to changed files.
- Inspect repository convention files, contribution docs, lint/config rules, and nearby existing code patterns before judging the implementation.
- Verify the PR follows the repo's established architecture, naming, error handling, testing, styling, and abstraction conventions.
- Prioritize actionable findings: bugs, regressions, security issues, data-loss risks, race conditions, edge cases, missing validation, broken API contracts, and insufficient tests.
- Check whether the code introduces unnecessary duplication, weak abstractions, over-engineering, or behavior that should reuse existing helpers/components.
- Look for security and privacy issues, including unsafe input handling, auth/permission gaps, secret exposure, injection risks, SSRF/path traversal/XSS/CSRF where relevant, and overly broad logging.
- Review tests for meaningful coverage of the changed behavior, including failure paths and important edge cases.
- Call out compatibility or migration risks if schemas, APIs, configs, background jobs, or user-facing behavior changed.

Please structure the review like this:
1. Findings first, ordered by severity.
2. For each finding, include file/line references, the concrete risk, and a suggested fix.
3. If there are no blocking issues, say that explicitly.
4. Include any open questions or assumptions.
5. Keep summary brief and avoid restating the diff.

Do not leave style-only comments unless they conflict with repository conventions or materially affect maintainability.

**Comment by findolor** [comment]
@coderabbitai review

File: src/approval_policy.rs [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+        condition
+            .split("&&")
+            .map(|term| policy_term_applicability(term, target)),
+    )
+}
+
+fn compact_expression(expression: Option<&str>) -> Option<String> {
+    let compact = expression?
+        .chars()
+        .filter(|character| !character.is_ascii_whitespace())
```
Comment (0xgleb) [inline src/approval_policy.rs]: minor: this removes whitespace inside quoted literals as well as syntax whitespace. A policy such as `eth.tx.to == '0x1111...1111 '` can therefore be certified as matching if Turnkey evaluates the trailing space as part of the literal and will reject the runtime activity

**Review by 0xgleb — APPROVED** [review]
